src/main/java/com/mabl/integration/jenkins/MablRestApiClientImpl.java

```diff
@@ -153,8 +153,15 @@ private CredentialsProvider getApiCredentialsProvider(
             final Secret restApiKey
     ) {
         final CredentialsProvider provider = new BasicCredentialsProvider();
+
+        // Adding check for null or empty restApiKey
+        final String keyTest = restApiKey.getPlainText();
+        if (keyTest == null || keyTest.trim().isEmpty()) {
+            throw new IllegalArgumentException ("API key not found");
+        }
```

Comment:
Curious, what is the UX around if this API key is `null`? Will the user see something in:
- Job run logs if it caused a failed job run
- UI if this is during an Applications/Labels/Environments API fetch call?

Finally, do we know this means there is no such secret in the Jenkins server? I'd think that would fail before the job could be run, or mabl code could be called. Perhaps this just means that the value is blank, and that users should double check it? It would be helpful to tell the user what's causing the issue, so they can easily correct it.

Reply:
I am not too sure about how this error would be floated up to the UI. That's part of what I am looking into for the logging improvements, but it's a twisty path to follow.

Reply:
@twistedpair Not sure how to test this, I only started on it because of ticket 6909. Any suggestions on moving this along?